src/main/java/com/aws/greengrass/clientdevices/auth/iot/usecases/VerifyIotCertificate.java

```diff
@@ -0,0 +1,92 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.iot.usecases;
+
+import com.aws.greengrass.clientdevices.auth.api.Result;
+import com.aws.greengrass.clientdevices.auth.api.UseCases;
+import com.aws.greengrass.clientdevices.auth.exception.CloudServiceInteractionException;
+import com.aws.greengrass.clientdevices.auth.infra.NetworkState;
+import com.aws.greengrass.clientdevices.auth.iot.Certificate;
+import com.aws.greengrass.clientdevices.auth.iot.IotAuthClient;
+import com.aws.greengrass.clientdevices.auth.iot.registry.CertificateRegistry;
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+
+import java.time.Instant;
+import java.util.Optional;
+import javax.inject.Inject;
+
+public class VerifyIotCertificate implements UseCases.UseCase<Exception, String> {
+    private static final Logger logger = LogManager.getLogger(VerifyIotCertificate.class);
+
+    private final NetworkState networkState;
+    private final IotAuthClient iotAuthClient;
+    private final CertificateRegistry certificateRegistry;
+
+
+    /**
+     * Register core certificate authority with Greengrass cloud.
+     *
+     * @param networkState        Network state infrastructure
+     * @param iotAuthClient       IoT dataplane API client
+     * @param certificateRegistry cert registry
+     */
+    @Inject
+    public VerifyIotCertificate(NetworkState networkState, IotAuthClient iotAuthClient,
+                                CertificateRegistry certificateRegistry) {
+        this.networkState = networkState;
+        this.iotAuthClient = iotAuthClient;
+        this.certificateRegistry = certificateRegistry;
+    }
+
+    // TODO: We need to be able to differentiate between a failure due
+    //  to cloud connectivity issues, and failures due to the certificate
+    //  not being active. The latter should result in us revoking the cert
+    //  from the registry, while the former means we should just fall back
+    //  to whatever is currently in the registry.
+    private Optional<String> getCertificateIdFromIot(String certificatePem) {
+        Optional<String> certificateId = Optional.empty();
+        try {
+            certificateId = iotAuthClient.getActiveCertificateId(certificatePem);
+        } catch (CloudServiceInteractionException e) {
+            // Do nothing
+        }
+        return certificateId;
+    }
+
+    @Override
+    public Result apply(String certificatePem) {
+        // If we think we are online, the certificate is not ACTIVE, or if certificate
+        // status information is stale, then go to the cloud.
+        // This ensures that we eventually go to IoT Core even if there are shenanigans
+        // going on where we have HTTP connectivity but MQTT is disconnected.
+        String source = "local";
+        Certificate cert = certificateRegistry.getCertificateByPem(certificatePem);
```

Comment:
Related to some of the comments above this would just internally compute the `id` from the pem file and try to get a certificate directly from the Repository and if not present create a new one like

```java
Optional<Certificate> cert = certificateRepository.getByPem(certificatePem);
// the isStale method would have cert.getStatus() != Certificate.Status.ACTIVE || !isTimestampRecent(cert) check

if (networkState.isUp() || certificate.isStale()) {
          // This would return a certificate that has source "cloud"
          Optional<Certificate> Certificate = getCertificateIdFromIot(certificatePem);
         // etc etc...
}
```

The main problem I see with this is that the Certificate object is not encapsulating business logic we have scattered methods that determine its logic across the Registry and this useCase